contracts/button/lib.rs

```diff
@@ -212,24 +228,11 @@ mod button_game {
         /// Can only be called after button's deadline
         #[ink(message)]
         pub fn reset(&mut self) -> ButtonResult<()> {
-            if !self.is_dead() {
-                return Err(GameError::BeforeDeadline);
-            }
-
-            let now = self.env().block_number();
+            self.ensure_dead()?;
+            self.reward_pressiah()?;
+            self.reset_state();
+            self.dump_tickets_in_marketplace()?;
```

Comment:
no need to return Ok(())
```suggestion
            self.dump_tickets_in_marketplace()
```